skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/SettingsActivity.java

```diff
@@ -48,21 +54,33 @@
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+
         super.onCreate(savedInstanceState);
         ViewGroup root = getRootView();
         Toolbar toolbar = (Toolbar) View.inflate(this, R.layout.toolbar, null);
         toolbar.setTitle(getString(R.string.settings));
         root.addView(toolbar, 0);
+        toolbar.setNavigationIcon(R.drawable.ic_back_navigation);
+        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    Intent intent = new Intent(SettingsActivity.this, MainActivity.class);
```

Comment:
You are giving intent here, it is not actually the good practice because if the user click back once again it will come to `settingsActivity` 
please try this 
```
        toolbar.setNavigationIcon(getResources().getDrawable(R.drawable.back_arrow));
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
```